ANOVA: fixed post-hoc tests  + descriptive plots

raviselker opened the pull request:
- Added support for interactions in post-hoc tests
- Added support for separate plots in descriptive plots
- Added option 'none' to the error bar defintion
- Added descriptives plots to non-parametric RM ANOVA

raviselker committed:
ANOVA: fixed post-hoc tests  + descriptive plots

- Added support for interactions in post-hoc tests
- Added support for separate plots in descriptive plots
- Added option 'none' to the error bar defintion

raviselker committed:
NP RM ANOVA: added descriptives plots

jonathon-love commented on R/anova.b.R:
```diff
@@ -256,73 +248,182 @@ anovaClass <- R6::R6Class(
+                table$addColumn(name='t', title='t', type='number')
+                
+                if (pnone)
+                    table$addColumn(name='pnone', title='p', type='number', format='zto,pvalue')
```
so i think you're better off *always* adding these, and toggling their visibility with the option.

calculating these additionally will add maybe 10ms to the analysis, but means that when someone toggles this option, a round trip to R won't be necessary (which will be super-quick once i implement that properly).

raviselker replied:
So what do you mean with "toggling their visibility with the option"?

jonathon-love replied:
using the data-binding. always calling `addColumn()`, but specifying visible as `(postHocCor:none)`, like in ... anova:

https://github.com/jamovi/jmv/blob/master/jamovi/anova.r.yaml#L170

(obviously i've specified it here in the yaml, but you can specify it in the .b.R too)

raviselker replied:
ok, got it

jonathon-love commented on R/anova.h.R:
```diff
@@ -135,6 +134,7 @@ anovaOptions <- R6::R6Class(
+                    "none",
                     "ci",
                     "se"),
                 default="ci")
```
cole reckons we should have 'none' as default. thoughts?

raviselker replied:
Isn't it better to have some kind of correction as default? Using normal p-values is pretty bad for post-hoc tests (actually using any type of inference on exploratory analyses is pretty silly).

jonathon-love replied:
for the plot, not p-values

raviselker replied:
hmm, apparently I wasn't really awake when I wrote the previous reply. I think a CI is pretty nice right, because it shows immediately which effect you can take seriously and which not.

jonathon-love replied:
i'm easy. i don't mind either way.

raviselker replied:
So let's leave the default on "ci" for now

jonathon-love commented on R/anovarmnp.b.R:
```diff
@@ -18,7 +18,7 @@ anovaRMNPClass <- R6::R6Class(
             mat[,i] <- jmvcore::toNumeric(data[[name]])
         }
-
+        
```
what's this!

raviselker replied:
What do you mean?

jonathon-love replied:
so this empty line, has been replaced with an empty line with whitespace in it. we try and avoid whitespace at the end of a line (or an empty line). in fact, the RStudio project file should have prevented this.

raviselker committed:
NP RM ANOVA: got rid of white space

raviselker committed:
ANOVA's: improved handling post-hoc tests

- Always calculate different corrections for post tests for ANOVA and RM
ANOVA
- Added no correction option to RM ANOVA
